### ISSUE TITLE
How to get the best mAP after the training completes

### ISSUE THREAD

**Issue by Chitti21** [tensorflow/models#10193]
@vighneshbirodkar Can you pls help with this?
How can I get  the best weights after the training completes?

For example, the graph below shows the total loss trained upto 30k. It can be noticed that the loss at 30k is not the global minimum of the graph. However, if I evaluate the metric using --check_point_dir argument, it will evaluate the metric at 30k only and not at global minimum.

![image](https://user-images.githubusercontent.com/75448105/128640871-4bd845ef-accb-4cfd-9eb8-43966efaaa6c.png)

**Is there an option to calculate mAP at the global minimum of the entire training session?**

**Comment by kumariko** [comment]
@Chitti21 In order to expedite the trouble-shooting process, Can you please provide the entire URL of the repository which you are using. Thanks!

**Comment by Chitti21** [comment]
@kumariko Thanks you for the response. 

URL that I cloned : https://github.com/tensorflow/models.git

**Comment by vighneshbirodkar** [comment]
@Chitti21 The lowest point in the training loss curve is not the global minimum because it is only evaluated on a single batch and not the entire training set. With that said, there is currently no way of doing what you are suggesting, but we would welcome a PR if you want to implement that.

**Comment by Chitti21** [comment]
@vighneshbirodkar Thank you for the information. This will be a very useful feature. I will try to contribute in the future. Closing this issue for now.
